fs: littlefs: add littlefs disk version selection

MikhailSiomin opened the pull request:
Add the ability to select littlefs disk version to maintain backward compatibility with existing littlefs with the same major disk version.

de-nordic reviewed (changes requested):
Change in LittleFS got NACK, there is no need to make change there.
The macros are only needed here.
Zephyr module of littlefs is synchronized from upstream LittleFS and changes done in our module are not required for it to work with Zephyr.

I do not mind change in the LittleFS Zephyr support code, this seems ok.

MikhailSiomin commented on subsys/fs/littlefs_fs.c:
```diff
@@ -882,6 +890,12 @@ static int littlefs_init_cfg(struct fs_littlefs *fs, int flags)
 
+#ifdef CONFIG_FS_LITTLEFS_DISK_VERSION
+	lcp->disk_version = disk_version;
+	LOG_INF("partition disk version: %u.%u", LFS_DISK_VERSION_MAJOR,
```
I think it's better to change `LFS_DISK_VERSION_MAJOR` by `FS_LITTLEFS_DISK_VERSION_MAJOR_GET(disk_version)`.

de-nordic replied:
I also think so, you may also report here what is actual driver supported version.

MikhailSiomin replied:
@de-nordic, the `littlefs_init_fs` function reports the actual supported disk version. Will this be enough?

de-nordic replied:
Yes.

MikhailSiomin committed:
fs: littlefs: add littlefs disk version selection

Add the ability to select littlefs disk version
to maintain backward compatibility
with existing littlefs
with the same major disk version.

Signed-off-by: Mikhail Siomin <victorovich.01@mail.ru>

MikhailSiomin commented:
@decsny, @galak, @Laczen, @nashif, can you please review this PR?
